src/init.cpp

```diff
@@ -1448,6 +1438,27 @@ bool AppInitParameterInteraction()
         InitWarning(_("-masternode option is deprecated and ignored, specifying -masternodeblsprivkey is enough to start this node as a masternode."));
     }
 
+    if (gArgs.IsArgSet("-masternodeblsprivkey")) {
+        if (!gArgs.GetBoolArg("-listen", DEFAULT_LISTEN)) {
```

Comment:
this seems error prone in case we ever change these defaults.

Also, my thought process was that we could wrap 
```
        gArgs.ForceSetArg("-listen", "1");
        LogPrintf("%s: parameter interaction: -masternodeblsprivkey=... -> setting -listen=1\n", __func__);
```
in a `if gArgs.GetArg("-listen", 1)`

Reply:
This specific part doesn't actually depend on the default value in any way. If we ever change defaults we should consider adding corresponding `SoftSetArg` in `InitParameterInteraction` instead.

Reply:
I don't understand how this doesn't depend on the default value

Reply:
Let me rephrase it. This reads `-listen` from config/cmd-line options and _if none was found_ assigns the value we specified (in this case `DEFAULT_LISTEN`). But in any case, we expect it to be `false` here regardless of the (default) value. If we ever change defaults to `false`, we should add
```
if (gArgs.IsArgSet("-masternodeblsprivkey")) {
    if (gArgs.SoftSetBoolArg("-listen", true)) {
        LogPrintf(...);
    }
}
```
in `InitParameterInteraction`. But this `if (!gArgs.GetBoolArg("-listen", DEFAULT_LISTEN)) {` code here should be kept untouched.

Reply:
if `-listen` hasn't been assigned, wouldn't it already be `DEFAULT_LISTEN`?

Reply:
I probably don't quite understand the issue I guess... Yes, if `-listen` hasn't been assigned yet, `GetBoolArg` should return `DEFAULT_LISTEN`. But it doesn't matter what the value of `DEFAULT_LISTEN` is, the code here is going to be the same both for `DEFAULT_LISTEN = true` and for `DEFAULT_LISTEN = false`. It's that code in `InitParameterInteraction` I mentioned above that should appear if `DEFAULT_LISTEN` would not match the expected result of `GetBoolArg` here.

Reply:
Looking at it again, I think this is good.